scylla-cql/src/types/deserialize/value.rs

```diff
@@ -114,10 +115,97 @@ fn mk_deser_err_named(
 /// Describes why serialization of some of the built-in types has failed.
 #[derive(Debug)]
 #[non_exhaustive]
-pub enum BuiltinDeserializationErrorKind {}
+pub enum BuiltinDeserializationErrorKind {
+    /// A generic deserialization failure - legacy error type.
+    GenericParseError(ParseError),
+}
 
```

Comment:
Is this only used for tests? Maybe it's changed in later commits, but I see it is introduced for tests right now.
If so, shouldn't it be a different error as this one is useless for the user? I thought that was the idea behind `DeserializationError` being dynamic: built-in errors only need to cover errors that may be returned by default implementations, and different implementations (like tests) can return different errors.

Reply:
The problem is that we we want to have a `BuiltinDeserializationError` for proper printing of `rust_name` and `got`, but for this we need to have a concrete `BuiltinDeserializationErrorKind` in hand. To achieve this in cases when we have a `ParseError` in hand, the `GenericParseError` variant is introduced.